### DISCUSSION TITLE
How to query the same object from different queries and make them same type?

### DISCUSSION THREAD

**Discussion by pjwalstrom** [ExpediaGroup/graphql-kotlin#1039] [Q&A] [answered]
Hello,
I have an external schema on which I have no possibility to make changes. The schema may look like this

```
type Query {
    foo: Bar
}
type Bar {
    qux: String
}
```

Then I have two different queries, let's say `MyFirstQuery.graphql` and `MySecondQuery.graphql`, that look the same
```
query {
    foo {
        qux
    }
}
```
When I generate the sources, I end up with two kotlin classes, `MyFirstQuery` and `MySecondQuery`, that both encapsulate the `data class Bar`. Which essentially is the same class, but the type system makes them different.

Is there a way to make `MyFirstQuery.Bar::class == MySecondQuery.Bar::class`?

I guess I could have done it by using interfaces in the schema, but I am not able to do that. Is there any other way, through querying?

Reference: https://github.com/pjwalstrom/GraphQLKotlinIssue

**Comment by dariuszkuc** [comment, marked as answer]
Currently there is no way to force the client generator to re-use the classes. As output classes can have different selection sets we followed the simple rule of always recreating them. 

I think we probably could update the logic to re-use common output types and input types (as input types will ALWAYS be the same across the queries).

**Comment by ermadmi78** [comment]
Kobby plugin generates DSL client that provides an ability to reuse generated classes
https://github.com/ermadmi78/kobby
